src/components/inputs/text-input/text-input.js

```diff
@@ -1,56 +1,95 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import requiredIf from 'react-required-if';
-import filterDataAttributes from '../../../utils/filter-data-attributes';
-import Constraints from '../../constraints';
+import isNil from 'lodash/isNil';
+import omit from 'lodash/omit';
 import { getInputStyles } from '../styles';
+import { getConstraintSyles } from '../../constraints/horizontal';
+import throwDeprecationWarning from '../../../utils/warn-deprecated-prop';
 
-const TextInput = props => (
-  <Constraints.Horizontal constraint={props.horizontalConstraint}>
-    <input
-      id={props.id}
-      name={props.name}
-      type="text"
-      value={props.value}
-      onChange={props.onChange}
-      onBlur={props.onBlur}
-      onFocus={props.onFocus}
-      disabled={props.isDisabled}
-      placeholder={props.placeholder}
-      readOnly={props.isReadOnly}
-      autoFocus={props.isAutofocussed}
-      autoComplete={props.autoComplete}
-      css={theme => getInputStyles(props, theme)}
-      {...filterDataAttributes(props)}
+const TextInput = props => {
+  function getInputProps(componentProps = {}) {
+    const readOnly = componentProps.readOnly || componentProps.isReadOnly;
+
+    return {
+      type: 'text',
+      readOnly,
+      disabled: componentProps.disabled || componentProps.isDisabled,
+      autoFocus: componentProps.autoFocus || componentProps.isAutofocussed,
       /* ARIA */
-      aria-readonly={props.isReadOnly}
-      role="textbox"
-      contentEditable={!props.isReadOnly}
+      role: 'textbox',
+      'aria-readonly': readOnly,
+      contentEditable: readOnly,
+      ...omit(componentProps, [
+        'hasError',
+        'hasWarning',
+        'horizontalConstraint',
+        /* deprecated */
+        'isReadOnly',
+        'isDisabled',
+        'isAutofocussed',
+      ]),
+    };
+  }
+
+  return (
+    <input
+      css={theme => [
+        getInputStyles(props, theme),
```

Comment:
couldn't we use the result of getInputProps here and then we wouldn't need to do the or checks?

Reply:
You mean the `or` checks inside the `getInputStyles`?
The problem is: other components, which haven't been refactored yet, also use `getInputStyles`, so it still needs to support the old names.

Reply:
ah yeah 🤔 makes sense

Reply:
Also, props such as `hasWarning` would be filtered out but they are needed by `getInputStyles`.